### DISCUSSION TITLE
Reject request by webhook

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#14145] [Feature Requests]
hi,team:

I am think how to support rejecting a request from webhook for custom validation logic. the webhook module just sends an HTTP request to a remote URL and ignored exceptions and responses in current. It's meaning `I just tell you once, and don't care your feeling`.

how about the new logic like this:

```js
try{
  const tx = db_transaction.begin();

  var data = tx.insertData(request);
  var resp = triggerWebhook("event.before_create",data)

  if(resp.status != 200) {
    tx.rollback();
    throw "webhook fails"
  }
} catch(e) {
    tx.rollback();
    throw "webhook fails"
}

db_transaction.commit();
```

**Comment by rijkvanzanten** [comment]
This can be done by using Flows rather than Webhooks 🙂

**Comment by imiskolee** [comment]
@rijkvanzanten  I tried the flows, but looks like can not work also, because when I set a webhook behind a flow's condition. the record also created when the webhook fails(can not connect / timeout/status code is not 200 ...).

**Reply by rijkvanzanten** [reply]
What are you trying to do exactly? Run a request on a filter, and prevent the item from being saved when the webhook doesn't succeed?

**Reply by imiskolee** [reply]
yes. 

example:

1. if a collection includes a field called `id_card_number`, we usually need to call a third-party API for verifying the field.

**Comment by imiskolee** [comment]
please take a look this idea: https://github.com/directus/directus/pull/14268

**Reply by rijkvanzanten** [reply]
Webhooks run on _actions_ async, so you can't prevent an item from being saved in webhooks

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
